widgetry/src/widgets/mod.rs

```diff
@@ -701,3 +703,32 @@ impl Widget {
         *self.widget.downcast::<JustDraw>().ok().unwrap()
     }
 }
+
+pub struct EdgeInsets {
+    pub top: f32,
+    pub left: f32,
+    pub bottom: f32,
+    pub right: f32,
+}
+
+impl From<usize> for EdgeInsets {
+    fn from(uniform_size: usize) -> EdgeInsets {
```

Comment:
note: I've retained the old API, where you specify uniform padding for all edges, which is a common case.